src/corehost/cli/version.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef __VERSION_H__
+#define __VERSION_H__
+
+#include "pal.h"
+#include "utils.h"
+
+struct version_t
```

Comment:
It would be good to document why we now have 2 "version" types.

Are we sure we can't use the existing fx_ver class?

Reply:
I added comments to version_t something like this:
```
// This type is made to match semantics of System.Version used by tooling and differs from ver_t:
// -- version_t does not require the third segment
// -- version_t does not support the "pre" label behavior
// -- Different terminology; fx_ver_t(major, minor, patch, build) vs version_t(major, minor, build, revision)
```